src/CollisionDetection.hs

```diff
@@ -1,103 +1,98 @@
 module CollisionDetection
   ( wallsCollision
   , bricksCollision
+  , paddleCollision
   , CollisionSide (..)
   ) where
 
 
-  import GameBoard
+import GameBoard
 
+-- | If there a collision on which side it accured
+data CollisionSide =
+      TopSide | BottomSide | LeftSide | RightSide
+      deriving Show
 
-  -- | If there a collision on which side it accured
-  data CollisionSide =
-    TopSide | BottomSide | LeftSide | RightSide
-    deriving Show
+-- | Given position and radius of the ball, return nothing if there is
+--   no collision or the collisionSide.
+wallsCollision :: Position              -- ^ ball position
+               -> Radius                -- ^ ball radius
+               -> Width                 -- ^ game width
```

Comment:
Personal taste: having a type alias per argument might be a tad too much. But it surely helps understanding what each argument means.